Source/WebCore/dom/Document.cpp

```diff
@@ -403,7 +404,18 @@ static void CallbackForContainIntrinsicSize(const Vector<Ref<ResizeObserverEntry
             ASSERT(box->style().hasAutoLengthContainIntrinsicSize());
 
             auto contentBoxSize = entry->contentBoxSize().at(0);
-            target->setLastRememberedSize(ResizeObserverSize::create(contentBoxSize->inlineSize(), contentBoxSize->blockSize()));
+            if (box->style().containIntrinsicWidthType() == ContainIntrinsicSizeType::AutoAndLength) {
+                if (box->isHorizontalWritingMode())
+                    target->setLastRememberedLogicalWidth(LayoutUnit(contentBoxSize->inlineSize()));
+                else
+                    target->setLastRememberedLogicalHeight(LayoutUnit(contentBoxSize->blockSize()));
+            }
+            if (box->style().containIntrinsicHeightType() == ContainIntrinsicSizeType::AutoAndLength) {
+                if (box->isHorizontalWritingMode())
+                    target->setLastRememberedLogicalHeight(LayoutUnit(contentBoxSize->blockSize()));
+                else
+                    target->setLastRememberedLogicalWidth(LayoutUnit(contentBoxSize->inlineSize()));
+            }
```

Comment:
Might be more readable if you add

```cpp
ContainIntrinsicSizeType containIntrinsicLogicalWidthType() const { return isHorizontalWritingMode() ? containIntrinsicWidthType() : containIntrinsicHeightType(); }
ContainIntrinsicSizeType containIntrinsicLogicalHeightType() const { return isHorizontalWritingMode() ? containIntrinsicHeightType() : containIntrinsicWidthType(); }
```

and then just

```cpp
if (box->style().containIntrinsicLogicalWidthType() == ContainIntrinsicSizeType::AutoAndLength)
    target->setLastRememberedLogicalWidth(LayoutUnit(contentBoxSize->inlineSize()));
if (box->style().containIntrinsicLogicalHeightType() == ContainIntrinsicSizeType::AutoAndLength) {
    target->setLastRememberedLogicalHeight(LayoutUnit(contentBoxSize->blockSize()));
```

Reply:
Done, and yes, it looks much better:)